src/python-bindings/htcondor2/submit.cpp

```diff
@@ -438,7 +438,11 @@ _submit_keys( PyObject *, PyObject * args ) {
     SubmitBlob * sb = (SubmitBlob *)handle->t;
     sb->keys(buffer);
 
-    return PyUnicode_FromStringAndSize( buffer.c_str(), buffer.size() - 1 );
+    if( buffer.size() == 0 ) {
+        Py_RETURN_NONE;
+    } else {
+        return PyUnicode_FromStringAndSize( buffer.c_str(), buffer.size() - 1 );
```

Comment:
We're 8-bit clean and only generate ASCII, so all of the strings we generate are valid UTF-8 unless some user really hates themselves (e.g, supplies a UTF-16 -encoded string where both bytes are non-zero).  Because we're 8-bit clean (enough), admins can do things like put stop-sign emoji in hold messages.  However, Python strings are Unicode, and Python therefore requires _some_ encoding be specified when creating a Python `str` object -- even though we don't actually know what our encoding is.  The way the ASCII "decoder" works, specifying that the C-side string is ASCII means that the transfer to Python isn't 8-bit clean!

So I use this method, which assumes UTF-8, which is almost certainly the right thing to do in every case anyway.